model/src/main/java/org/openremote/model/teltonika/State.java

```diff
@@ -0,0 +1,298 @@
+                        //      It is calculated correctly, check with a debugger, but why is it not storing the data as required?
+                        if(multiplier != 1L){
+                            finalMultiplier = multiplier;
+                        }
+
+                        // Check if the value is correctly within the constraints. If it's not, don't apply the constraint.
+                        // The only reason I am doing this is that the constraints are currently not programmatically, thus "seriously", set.
+                        // If it was properly defined, then parameter ID 237 wouldn't be inaccurate, let alone to this state.
+                        // Not to mention parsing errors (from example from the UDP/Codec 8 to MQTT/Codec JSON converter, look at accelerator axes)
+                        // THE AXES VARS OVERFLOW  - if i see that TCT has value -46, if I do 65535 minus the value I am given, then it gives me the real value
+                        if (min.isPresent()) {
+                            if ( !((Double) value.get() < (Double) min.get())) {
+                                constraintValues.add(new ValueConstraint.Min((Double) min.get() * finalMultiplier));
+                            }
+                        }
+                        if(max.isPresent()){
+                            if ( !((Double) value.get() > (Double) max.get())) {
+                                constraintValues.add(new ValueConstraint.Max((Double) max.get() * finalMultiplier));
+                            }
+                        }
+
+                        ValueConstraint[] constraints = constraintValues.toArray(new ValueConstraint[0]);
+
+                        constraintsMeta.setValue(constraints);
+
+                        //TODO: Fix this, constraints for some reason are not being applied
+//                        metaMap.add(constraintsMeta);
+                    }
+                }catch (Exception e){
+                    logger.severe(parameter.propertyName + "Failed constraints");
+                    logger.severe(e.toString());
+                    throw e;
+                }
+            }
+            // Add on its label
+            try{
+                MetaItem<String> label = new MetaItem<>(MetaItemType.LABEL);
+                label.setValue(parameter.propertyName);
+                metaMap.add(label);
+            }catch (Exception e){
+                logger.severe(parameter.propertyName + "Failed label");
+                logger.severe(e.toString());
+                throw e;
+            }
+            //Use the MetaMap to create an AttributeDescriptor
+            AttributeDescriptor<?> attributeDescriptor = new AttributeDescriptor<>(parameter.propertyId.toString(), attributeType, metaMap);
+
+
+
+            //Use the AttributeDescriptor and the Value to create a new Attribute
+            Attribute<?> generatedAttribute = new Attribute(attributeDescriptor, value.get());
```

Comment:
`AttributeDescriptor` are for static use not runtime use; you should just use the other constructor(s):

```
Attribute(String name, ValueDescriptor<T> valueDescriptor, T value)
```